recognition/advanced_tree_search.py

```diff
@@ -855,19 +862,23 @@ def cleanup_before_run(self, cmd, retry, *args):
         util.backup_if_exists("build_global_cache.log")
 
     @classmethod
-    def create_config(cls, crp, extra_config, extra_post_config, **kwargs):
-        config, post_config = rasr.build_config_from_mapping(
-            crp,
-            {
-                "lexicon": "speech-recognizer.model-combination.lexicon",
-                "acoustic_model": "speech-recognizer.model-combination.acoustic-model",
-                "language_model": "speech-recognizer.model-combination.lm",
-                "recognizer": "speech-recognizer.recognizer",
-            },
-        )
+    def create_config(cls, crp, search_type, recognizer_type, extra_config, extra_post_config, **kwargs):
+        mapping_dict = {
+            "lexicon": "speech-recognizer.model-combination.lexicon",
+            "acoustic_model": "speech-recognizer.model-combination.acoustic-model",
+            "language_model": "speech-recognizer.model-combination.lm",
+            "recognizer": "speech-recognizer.recognizer",
+        }
+
+        if recognizer_type == "recognizer-v2":
+            mapping_dict["label_scorer"] = "speech-recognizer.model-combination.label-scorer"
+
+        config, post_config = rasr.build_config_from_mapping(crp, mapping_dict)
 
         config.speech_recognizer.recognition_mode = "init-only"
-        config.speech_recognizer.search_type = "advanced-tree-search"
+        config.speech_recognizer.search_type = search_type
+        if recognizer_type != "recognizer":
+            config.speech_recognizer.reocgnizer_type = recognizer_type
```

Comment:
This does not work. The speech-recognizer doesn't know anything about recognizer-v2. Right now it uses this method to create the recognizer: https://github.com/rwth-i6/rasr/blob/master/src/Search/Module.cc#L77 (called from https://github.com/rwth-i6/rasr/blob/master/src/Tools/SpeechRecognizer/SpeechRecognizer.cc#L114 )

Also note that the new recognizer uses a parameter named "type" and not "search-type" to determine the exact algorithm. This also has to change.

Reply:
Here is the i6 RASR PR: https://github.com/rwth-i6/rasr/pull/142

Reply:
And `search_type` is actually not used in the speech-recognizer. 
But deleting this will result in hash break so I'll leave this..